library/std/src/panicking.rs

```diff
@@ -44,11 +44,10 @@ use realstd::io::set_output_capture;
 extern "C" {
     fn __rust_panic_cleanup(payload: *mut u8) -> *mut (dyn Any + Send + 'static);
 
-    /// `payload` is actually a `*mut &mut dyn BoxMeUp` but that would cause FFI warnings.
-    /// It cannot be `Box<dyn BoxMeUp>` because the other end of this call does not depend
-    /// on liballoc, and thus cannot use `Box`.
+    /// `payload` is actually a `Box<dyn BoxMeUp>`, but we pass this by-reference because the other
+    /// end of this call does not depend on liballoc, and thus cannot use `Box`.
```

Comment:
> `payload` is actually a `Box<dyn BoxMeUp>`

Oh, that doesn't sound correct either. None of the three `BoxMeUp` implementations contain a `Box` or are wrapped themselves in a `Box`. The objects implementing `BoxMeUp` are owned by the caller and stored on the stack.

I guess originally this was just to avoid using `Box` in core, but now it's also used to lazily allocate the `Box` only when needed, and to provide access the payload without putting it in a `Box` first.

(I suppose `&mut dyn BoxMeUp` instead of `*mut &mut dyn BoxMeUp` would work too, with the FFI warnings disabled.)

Reply:
All right, I updated the comment.